Model/Billing/Agreement.php

```diff
@@ -179,36 +179,41 @@ public function setCcBillingAgreement($contractDetail, $storeOneClick, $storeId)
             ->setMethodCode('adyen_oneclick')
             ->setReferenceId($contractDetail['recurring.recurringDetailReference']);
 
+        if (!isset($contractDetail['cardBin']) ||
+            !isset($contractDetail['cardHolderName']) ||
+            !isset($contractDetail['cardSummary']) ||
+            !isset($contractDetail['expiryDate']) ||
+            !isset($contractDetail['paymentMethod'])
+        ) {
+            $this->_errors[] = __('"In the Additional data in API response section, select: Card summary, Expiry Date, Cardholder name, Recurring details and Variant to create billing agreements immediately after the payment is authorized."');
```

Comment:
where is this being used ?

Reply:
We show it to the user here:
https://github.com/Adyen/adyen-magento2/pull/453/files#diff-735f70c1f948ebbd90915b1548d93432R1631

This is shown at the magento backend under the order

Reply:
thanks clear